.buildkite/pipelines/on_merge.yml

```diff
@@ -169,6 +169,25 @@ steps:
         - exit_status: '*'
           limit: 1
 
+  - command: .buildkite/scripts/steps/test/ftr_configs.sh
+    env:
+      FTR_CONFIG: "x-pack/test/functional_gen_ai/inference/config.ts"
+      FTR_CONFIG_GROUP_KEY: 'ftr-ai-infra-gen-ai-inference-api'
+      FTR_GEN_AI: "1"
+    label: AppEx AI-Infra Inference APIs FTR tests
+    key: ai-infra-gen-ai-inference-api
+    timeout_in_minutes: 50
+    parallelism: 1
+    agents:
+      machineType: n2-standard-4
+      preemptible: true
+    retry:
+      automatic:
+        - exit_status: '-1'
+          limit: 3
+        - exit_status: '*'
+          limit: 1
```

Comment:
We can always emit yml towards buildkite to create new steps.

This means, we could have a script, that points to a file, and uploads steps from that. Then invoke the script in a command step, like:
```yaml
 - command: .buildkite/scripts/upload_pipeline_steps.sh <path-to-your-yml>
   ...
```

Then the implementation could invoke `buildkite-agent` (like this, but in a bit nicer fashion):
```bash
# script: upload_pipeline_steps.sh

cat $1 | buildkite-agent pipeline upload
```

In the end, you'll still have to have an extra step in the on-merge pipeline, but at least you'd share the same yml file (.buildkite/pipelines/pull_request/ai_infra_gen_ai.yml) and remove the duplication.

Reply:
Got it. It's more nice to have rather than a requirement on my side though - and if we wanted to do it, I assume we'd like to do it for all the duplicated steps that we have in `.buildkite/pipelines/on_merge.yml`.

Reply:
good point. Would you leave this as an optimization step for us rather? To do all of these together in one PR? If so, I'll create an issue on our board

Reply:
Yeah if you don't mind, I think it would make more sense that way